unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -6936,129 +6936,149 @@ public bool objectIsCurrentlyVisible(SimObjPhysics sop, float maxDistance)
-            {
-                errorMessage = "Horizon value for PositionsFromWhichItemIsInteractable must be in range [0, 30] inclusive";
+        // Get the poses with which the agent can interact with 'objectId'
+        // @rotations: if rotation is not specified, we use rotateStepDegrees, which results in [0, 90, 180, 270] by default.
+        public void GetInteractablePositions(
+            string objectId,
+            Vector3[] positions = null,
+            float[] rotations = null,
+            float[] horizons = null,
+            bool[] standings = null
+        ) {
+            if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(objectId)) {
+                errorMessage = "objectId appears to be invalid.";
                 actionFinished(false);
                 return;
             }
-            Vector3[] positions = null;
-            if (action.positions != null && action.positions.Count != 0) {
-                positions = action.positions.ToArray();
-            } else {
-                positions = getReachablePositions();
-            }
+            SimObjPhysics theObject = physicsSceneManager.ObjectIdToSimObjPhysics[objectId];
 
+            // save current agent pose
             bool wasStanding = isStanding();
             Vector3 oldPosition = transform.position;
             Quaternion oldRotation = transform.rotation;
             Vector3 oldHorizon = m_Camera.transform.localEulerAngles;
+            if (ItemInHand != null) {
+                ItemInHand.gameObject.SetActive(false);
+            }
 
-            if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(action.objectId)) {
-                errorMessage = "Object ID appears to be invalid.";
-                actionFinished(false);
-                return;
+            // Populate default standings. Note that these are boolean because that's
+            // the most natural integration with Teleport
+            if (standings == null) {
+                standings = new bool[] {false, true};
             }
 
-            if (ItemInHand != null) {
-                ItemInHand.gameObject.SetActive(false);
+            // populate default horizons
+            if (horizons == null) {
+                horizons = new float[] {-30, 0, 30, 60};
             }
 
-            SimObjPhysics theObject = physicsSceneManager.ObjectIdToSimObjPhysics[action.objectId];
+            // populate the positions by those that are reachable
+            if (positions == null) {
+                positions = getReachablePositions();
+            }
+
+            // populate the rotations based on rotateStepDegrees
+            if (rotations == null) {
+                rotations = new float[(int) (360 / rotateStepDegrees)];
+                int i = 0;
+                for (float rotation = 0; rotation < 360; rotation += rotateStepDegrees) {
+                    rotations[i++] = rotation;
+                }
+            }
```

Comment:
Rotations are now dynamically set based on `rotateStepDegrees`. This has caused several silent bugs in the past when using non-90 degree rotations.

Reply:
A few thoughts:

* What if I teleport my agent to a rotation that's not on the standard `[0, rotateStepDegrees, 2*rotateStepDegrees, ...]` set of rotations. Does it make more sense to have these rotations be relative this non-standard rotation, i.e. do something like
```
offset = cur_rotation % rotateStepDegrees
rotations = [offset, offset + rotateStepDegrees, 2 * rotateStepDegrees, ...]
```
* It looks like `rotateStepDegrees` is allowed to be any possible floating point. If I pick `rotateStepDegrees=any irrational number` then the set `{[0, rotateStepDegrees % 360, 2 * rotateStepDegrees  % 360, ...}` will never repeat, thus does it really make sense to limit the rotations in the way you are? Perhaps we should only allow this if `(360 / rotateStepDegrees) is an integer and otherwise throw an error.

Reply:
For 1) I like the idea and would have the agent go through the full set of rotations from [offset : 360 + offset).

For instance:
* If `rotateStepDegrees=10` and `offset=70`, then the paths would be `[70, 80, ..., 400, 410, 420]`.
* If `rotateStepDegrees=90` and `offset=10`, then the paths would be `[10, 100, 190, 280]`

For 2) Seems reasonable to just give an error, since, for irrational rotations, users can override the `rotations` if they have something in mind.